franka_description/robots/arm.xacro

```diff
@@ -276,16 +256,19 @@
           <sphere radius="${0.04+safety_distance}"  />
         </geometry>
       </collision>
-    </link>
-    <joint name="${arm_id}_joint7" type="revolute">
+      </self_collision_geometries>
+    </xacro:link>
+    <joint name="${prefix}joint7" type="revolute">
       <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
       <origin rpy="${pi/2} 0 0" xyz="0.088 0 0"/>
-      <parent link="${arm_id}_link6"/>
-      <child link="${arm_id}_link7"/>
+      <parent link="${prefix}link6"/>
+      <child link="${prefix}link7"/>
       <axis xyz="0 0 1"/>
       <limit effort="12" lower="-2.8973" upper="2.8973" velocity="2.6100"/>
+      <dynamics D="1" K="7000" damping="0.003" friction="0.0" mu_coulomb="0" mu_viscous="16"/>
     </joint>
-    <link name="${arm_id}_link8">
+    <link name="${prefix}link8"/>
+    <link name="${prefix}link8_sc">
```

Comment:
Why did you introduce a new `link8_sc` and what does `sc` stand for?

Reply:
...probably self collision I guess. Now I see that your `<self_collision_geometries />` tag simply generates new links with a `_sc` suffix and the collision geometries beneath that. TBH I assumed that MoveIT would pick up the this tag and interpret it accordingly rather then looking for specifically named links...

Reply:
As explained in the main comment, I just introduce both, detailed and coarse collision models for each link - the latter ones only used for self-collision checking. MoveIt doesn't look for specific names. You can pick any name you want.
The semantics of the different collision models is eventually defined in the SRDF.